xds/src/main/java/io/grpc/xds/RingHashLoadBalancer.java

```diff
@@ -381,38 +393,78 @@ private int getTargetIndex(Long requestHash) {
 
     @Override
     public PickResult pickSubchannel(PickSubchannelArgs args) {
-      Long requestHash = args.getCallOptions().getOption(XdsNameResolver.RPC_HASH_KEY);
-      if (requestHash == null) {
-        return PickResult.withError(RPC_HASH_NOT_FOUND);
+      // Determine request hash.
+      boolean usingRandomHash = false;
+      Long requestHash;
```

Comment:
Let's not box and think about null in these new cases that don't need it. We should make this `long` and deal with the `null` in the only case that needs it. We should also change `getTargetIndex()` be be passed a `long` (since it assumes it is non-null already).

Reply:
`CallOptions` is immutable. To change the options you have to create a new instance. So there were no thread-safety or race issues. But getting the option only once is good and what I would have done.

Reply:
(And yes, pickSubchannel does not run in the syncContext so it needs to be thread-safe.)